### PR TITLE
Have P4ChangeRequestSCMHead implement interface ChangeRequestSCMHead2

### PR DESCRIPTION
**Pull request by dvilaverde**
Have the P4ChangeRequestSCMHead class implement ChangeRequestSCMHead2 interface so the environment variable CHANGE_BRANCH will have a valid P4 depot path provided by P4SCMHead set as the target for the P4ChangeRequestSCMHead.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/p4/scm/SwarmScmSource.java [hunk 1]
```diff
@@ -2,8 +2,11 @@
 
 import edu.umd.cs.findbugs.annotations.NonNull;
 import hudson.Extension;
+import hudson.model.Action;
 import hudson.model.TaskListener;
-import jenkins.scm.api.SCMHeadCategory;
+import jenkins.scm.api.*;
```
Comment (p4paul) [inline src/main/java/org/jenkinsci/plugins/p4/scm/SwarmScmSource.java:7]: Please avoid wildcard imports (for non-static)

**Comment by p4paul** [comment]
Thank you for the PR - I'll need to run this through our system tests, but the change looks good.

File: src/main/java/org/jenkinsci/plugins/p4/scm/SwarmScmSource.java [hunk 2]
```diff
@@ -125,7 +143,7 @@ public P4Browser getBrowser() {
 			// Get first Swarm path; it MUST include the Jenkinsfile
 			P4Path p4Path = branch.getPath();
 
-			P4SCMHead head = new P4SCMHead(branch.getId(), p4Path);
+			P4SCMHead head = new P4SCMHead(branch.getName(), p4Path);
```
Comment (p4paul) [inline src/main/java/org/jenkinsci/plugins/p4/scm/SwarmScmSource.java:146]: (reminder for me to check all uses of getId() for case sensitivity)
#PR124 addresses another instance of getId() / getName()

File: src/main/java/org/jenkinsci/plugins/p4/scm/SwarmScmSource.java (hunk 1 above)
Comment (dvilaverde) [reply src/main/java/org/jenkinsci/plugins/p4/scm/SwarmScmSource.java]: updated to remove the wildcard

**Pull request merged**